File: backend/app/schemas/ticket.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TicketBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: Optional[str] = "Open"
    priority: Optional[str] = "Medium"
    client_id: int


class TicketCreate(TicketBase):
    pass


class TicketRead(TicketBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TicketUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: str | None = None
    priority: str | None = None
    client_id: int | None = None

File: frontend/src/types/ticket.ts
export type Ticket = {
  id: number;
  title: string;
  description: string | null;
  status: string | null;
  priority: string | null;
  client_id: number;
  created_at: string;
};

export type TicketCreate = {
  title: string;
  description?: string | null;
  status?: string | null;
  priority?: string | null;
  client_id: number;
};

export type TicketUpdate = {
  title?: string | null;
  description?: string | null;
  status?: string | null;
  priority?: string | null;
  client_id?: number | null;
};
